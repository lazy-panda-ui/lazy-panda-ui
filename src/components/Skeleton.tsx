import React from 'react';
import { View, StyleSheet, ViewStyle, DimensionValue } from 'react-native';
import { useTheme } from '../theme/ThemeProvider';

export interface SkeletonProps {
  width?: DimensionValue;
  height?: DimensionValue;
  style?: ViewStyle;
  borderRadius?: number;
}

export const Skeleton: React.FC<SkeletonProps> = ({ width = '100%', height = 20, style, borderRadius }) => {
  const theme = useTheme();
  return (
    <View
      style={[
        styles(theme).skeleton,
        { width, height, borderRadius: borderRadius ?? theme.borderRadius } as ViewStyle,
        style,
      ]}
    />
  );
};

const styles = (theme: any) => StyleSheet.create({
  skeleton: {
    backgroundColor: theme.colors.border,
    opacity: 0.5,
  },
});
