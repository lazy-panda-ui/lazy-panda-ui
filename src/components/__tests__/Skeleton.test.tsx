import React from 'react';
import { render } from '@testing-library/react-native';
import { Skeleton } from '../Skeleton';

describe('Skeleton', () => {
  it('renders', () => {
    const { toJSON } = render(<Skeleton />);
    expect(toJSON()).toMatchSnapshot();
  });
});
