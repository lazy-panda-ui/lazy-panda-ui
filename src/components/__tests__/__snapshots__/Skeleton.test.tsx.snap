// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton renders 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#E5E5EA",
        "opacity": 0.5,
      },
      {
        "borderRadius": 8,
        "height": 20,
        "width": "100%",
      },
      undefined,
    ]
  }
/>
`;
